### PR TITLE
[Bench] Coverity Fixes

### PR DESCRIPTION
**Pull request by simpsont-oci**
Problem: Recent Coverity scans were complaining about a number of issues in Bench, particularly uncaught exceptions and a few resource locking / cleanup issues.

Solution: These changes should resolve a large number of the issues

### CODE REVIEW

File: performance-tests/bench/tcp_latency/tcp_latency.cpp
Comment (mitza-oci) [inline performance-tests/bench/tcp_latency/tcp_latency.cpp:246]:
```suggestion
    ACE_ERROR((LM_ERROR, "Unknown Exception Caught\n"));
```

I will leave it up to you to determine of Bench is following all of the development guidelines wrt logging

**Review by mitza-oci — CHANGES_REQUESTED** [review]

**Review by mitza-oci — APPROVED** [review]

File: performance-tests/bench/node_controller/ProcessStatsCollector.cpp
Comment (ClaytonCalabrese) [inline performance-tests/bench/node_controller/ProcessStatsCollector.cpp:218]:
```suggestion
      percent /= time - last_time_; // total cpu time is across all processors (ignore num_processors_)
```

**Review by ClaytonCalabrese — APPROVED** [review]
Just a single comment spelling error. I don't think it is worth blocking the merge over.

**Pull request merged**
